diff --git a/OpenCVModule.cmake b/OpenCVModule.cmake
index f7620b6..0eb2225 100644
--- a/OpenCVModule.cmake
+++ b/OpenCVModule.cmake
@@ -27,11 +27,13 @@ macro(define_opencv_module name)
     file(GLOB lib_hdrs "include/opencv2/${name}/*.h*")
     source_group("Include" FILES ${lib_hdrs})
 
-    set(the_target "opencv_${name}")
+	set(the_target "opencv_${name}")
+	set(the_target_s "opencv_${name}_s")
     if (${name} MATCHES "ts" AND MINGW)
         add_library(${the_target} STATIC ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs})
     else()
         add_library(${the_target} ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs})
+        add_library(${the_target_s} STATIC ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs})
     endif()
 
     # For dynamic link numbering convenions
diff --git a/modules/gpu/CMakeLists.txt b/modules/gpu/CMakeLists.txt
index a8cc0fe..b3118ef 100644
--- a/modules/gpu/CMakeLists.txt
+++ b/modules/gpu/CMakeLists.txt
@@ -1,6 +1,7 @@
 set(name "gpu")
 
 set(the_target "opencv_${name}")
+set(the_target_s "opencv_${name}_s")
 project(${the_target})
 
 set(DEPS "opencv_core" "opencv_imgproc" "opencv_objdetect" "opencv_features2d" "opencv_flann" "opencv_calib3d") #"opencv_features2d" "opencv_flann" "opencv_objdetect" - only headers needed 
@@ -84,7 +85,7 @@ foreach(d ${DEPS})
 endforeach()
 
 add_library(${the_target} ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs} ${lib_cuda} ${lib_cuda_hdrs} ${lib_device_hdrs} ${ncv_srcs} ${ncv_hdrs} ${ncv_cuda} ${cuda_objs})
-
+add_library(${the_target_s} STATIC ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs} ${lib_cuda} ${lib_cuda_hdrs} ${lib_device_hdrs} ${ncv_srcs} ${ncv_hdrs} ${ncv_cuda} ${cuda_objs})
 if(PCHSupport_FOUND)
 	set(pch_header ${CMAKE_CURRENT_SOURCE_DIR}/src/precomp.hpp)
 	if(${CMAKE_GENERATOR} MATCHES "Visual*" OR ${CMAKE_GENERATOR} MATCHES "Xcode*")
diff --git a/modules/highgui/CMakeLists.txt b/modules/highgui/CMakeLists.txt
index 2e12aa6..89d4110 100644
--- a/modules/highgui/CMakeLists.txt
+++ b/modules/highgui/CMakeLists.txt
@@ -245,6 +245,7 @@ set(lib_srcs ${highgui_srcs} ${grfmt_srcs})
 #               Define the library target:
 # ----------------------------------------------------------------------------------
 set(the_target "opencv_highgui")
+set(the_target_s "opencv_highgui_s")
 
 include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include"
                     "${CMAKE_CURRENT_SOURCE_DIR}/../core/include"
@@ -257,6 +258,7 @@ if(WIN32)
 endif()
 
 add_library(${the_target} ${lib_srcs} ${highgui_hdrs} ${grfmt_hdrs} ${highgui_ext_hdrs})
+add_library(${the_target_s} STATIC ${lib_srcs} ${highgui_hdrs} ${grfmt_hdrs} ${highgui_ext_hdrs})
 
 if (BUILD_SHARED_LIBS)
 	add_definitions(-DHIGHGUI_EXPORTS)
