--- a/CMakeLists.txt	2011-07-08 11:50:16.000000000 +0900
+++ b/CMakeLists.txt	2011-07-08 11:50:24.000000000 +0900
@@ -1023,7 +1023,7 @@
         if(NOT MINGW64)
             if(NOT X86_64)
                 if(NOT APPLE)
-                    set(EXTRA_C_FLAGS "${EXTRA_C_FLAGS} -march=i686")
+                    set(EXTRA_C_FLAGS "${EXTRA_C_FLAGS} -march=i486")
                 endif()
             endif()
         endif()
