--- a/interfaces/swig/python/Makefile.am	2009-11-26 08:39:51.000000000 +0900
+++ b/interfaces/swig/python/Makefile.am	2009-11-26 08:40:34.000000000 +0900
@@ -130,7 +130,7 @@
 _highgui_la_SOURCES    = _highgui.cpp pyhelpers.cpp pyhelpers.h \
                          highgui.i
 _highgui_la_CXXFLAGS     = $(PYTHON_CSPEC) $(AM_CXXFLAGS)
-_highgui_la_LIBADD       = $(top_builddir)/src/libhighgui.la
+_highgui_la_LIBADD       = $(top_builddir)/src/libhighgui.la $(top_builddir)/cv/libcv.la $(top_builddir)/src/libcxcore.la
 _highgui_la_LDFLAGS      = -module -avoid-version -no-undefined $(PYTHON_LSPEC) $(SWIG_PYTHON_LIBS) @LDFLAGS@
 
 endif
