Index: opencv-1.0.0/configure.in
===================================================================
--- opencv-1.0.0.orig/configure.in	2009-03-01 22:32:22.000000000 +0100
+++ opencv-1.0.0/configure.in	2009-03-01 22:32:30.000000000 +0100
@@ -483,14 +483,14 @@
                                      ]))
         
         if test x"$have_raw1394" = "xyes"; then
-            AC_CHECK_HEADER(libdc1394/dc1394_control.h,
-                            AC_CHECK_LIB(dc1394_control, dc1394_camera_on, 
+            AC_CHECK_HEADER(dc1394/control.h,
+                            AC_CHECK_LIB(dc1394, dc1394_camera_set_power, 
                                          [
                                            have_dc1394=yes
                                            AC_DEFINE(HAVE_DC1394,,[IEEE1394 capturing support])
-                                           AC_EGREP_HEADER(do_extra_buffering, libdc1394/dc1394_control.h,
-                                                           [AC_DEFINE(HAVE_DC1394_095,,[libdc1394 0.9.4 or 0.9.5])])
-                                           IEEE1394LIBS="-lm -lraw1394 -ldc1394_control"
+                                           AC_EGREP_HEADER(dc1394_camera_set_power, dc1394/control.h,
+                                                           [AC_DEFINE(HAVE_DC1394_2_X,,[libdc1394 2.x])])
+                                           IEEE1394LIBS="-lm -lraw1394"
                                          ]))
         fi
         
