Index: opencv-0.9.7/cxcore/include/cxtypes.h
===================================================================
--- opencv-0.9.7.orig/cxcore/include/cxtypes.h	2006-05-14 05:24:22.000000000 +0200
+++ opencv-0.9.7/cxcore/include/cxtypes.h	2006-05-14 05:24:25.000000000 +0200
@@ -115,7 +115,7 @@
 #elif (defined WIN32 || defined WIN64) && !defined __GNUC__
     #define CV_INLINE __inline
 #else
-    #define CV_INLINE static
+    #define CV_INLINE static inline
 #endif
 #endif /* CV_INLINE */
 
Index: opencv-0.9.7/otherlibs/highgui/highgui.h
===================================================================
--- opencv-0.9.7.orig/otherlibs/highgui/highgui.h	2005-07-05 17:52:50.000000000 +0200
+++ opencv-0.9.7/otherlibs/highgui/highgui.h	2006-05-14 05:24:25.000000000 +0200
@@ -83,7 +83,7 @@
     #elif (defined WIN32 || defined WIN64) && !defined __GNUC__
       #define CV_INLINE __inline
     #else
-      #define CV_INLINE static
+      #define CV_INLINE static inline
     #endif
   #endif /* CV_INLINE */
 
