--- a/include/opencv/cxtypes.h.orig	2009-11-25 16:35:30.000000000 +0900
+++ b/include/opencv/cxtypes.h	2009-11-25 16:37:25.000000000 +0900
@@ -78,7 +78,7 @@
 
   #if ((defined WIN32 || defined _WIN32 || defined WIN64 || defined _WIN64) && \
       (_MSC_VER >= 1400 || defined CV_ICC)) \
-      || (defined __SSE2__ && defined __GNUC__ && __GNUC__ >= 4)
+      || (defined __SSE2__ && defined __GNUC__ && __GNUC__ >= 4) || defined __x86_64__
     #include <emmintrin.h>
     #define CV_SSE2 1
   #else
