--- a/include/opencv/cxtypes.h	2009-11-25 16:30:59.000000000 +0900
+++ b/include/opencv/cxtypes.h	2009-11-25 16:31:15.000000000 +0900
@@ -140,7 +140,7 @@
 #elif (defined WIN32 || defined _WIN32 || defined WIN64 || defined _WIN64 || defined WINCE) && !defined __GNUC__
     #define CV_INLINE __inline
 #else
-    #define CV_INLINE static
+    #define CV_INLINE static inline
 #endif
 #endif /* CV_INLINE */
 
--- a/include/opencv/highgui.h	2009-11-25 16:32:04.000000000 +0900
+++ b/include/opencv/highgui.h	2009-11-25 16:33:47.000000000 +0900
@@ -83,7 +83,7 @@
     #elif (defined WIN32 || defined _WIN32 || defined WIN64 || defined _WIN64) && !defined __GNUC__
       #define CV_INLINE __inline
     #else
-      #define CV_INLINE static
+      #define CV_INLINE static inline
     #endif
   #endif /* CV_INLINE */
 
