diff --git a/apps/haartraining/cvboost.cpp b/apps/haartraining/cvboost.cpp
index 210a14e..037874b 100644
--- a/apps/haartraining/cvboost.cpp
+++ b/apps/haartraining/cvboost.cpp
@@ -64,6 +64,9 @@
 #include "_cvcommon.h"
 #include "cvclassifier.h"
 
+#ifdef _OPENMP
+#include "omp.h"
+#endif
 
 #define CV_BOOST_IMPL
 
diff --git a/include/opencv/cxmisc.h b/include/opencv/cxmisc.h
index d808c7e..c24cd95 100644
--- a/include/opencv/cxmisc.h
+++ b/include/opencv/cxmisc.h
@@ -48,14 +48,7 @@
 #ifndef _CXCORE_MISC_H_
 #define _CXCORE_MISC_H_
 
-#ifdef HAVE_CONFIG_H
-    #include "cvconfig.h"
-#endif
-
 #include <limits.h>
-#ifdef _OPENMP
-#include "omp.h"
-#endif
 
 /****************************************************************************************\
 *                              Compile-time tuning parameters                            *
diff --git a/src/cvaux/vs/blobtrackingmsfg.cpp b/src/cvaux/vs/blobtrackingmsfg.cpp
index e275535..8b74554 100644
--- a/src/cvaux/vs/blobtrackingmsfg.cpp
+++ b/src/cvaux/vs/blobtrackingmsfg.cpp
@@ -40,6 +40,10 @@
 
 #include "_cvaux.h"
 
+#ifdef _OPENMP
+#include "omp.h"
+#endif
+
 // Uncomment to trade flexibility for speed
 //#define CONST_HIST_SIZE
 
diff --git a/src/cxcore/cxsystem.cpp b/src/cxcore/cxsystem.cpp
index 0cdbe21..c69e72f 100644
--- a/src/cxcore/cxsystem.cpp
+++ b/src/cxcore/cxsystem.cpp
@@ -50,6 +50,10 @@
 #include <time.h>
 #endif
 
+#ifdef _OPENMP
+#include "omp.h"
+#endif
+
 #include <stdarg.h>
 
 namespace cv
-- 
1.6.6

