Index: opencv-1.0.0/cxcore/include/cxtypes.h
===================================================================
--- opencv-1.0.0.orig/cxcore/include/cxtypes.h	2009-11-13 16:12:26.000000000 +0100
+++ opencv-1.0.0/cxcore/include/cxtypes.h	2009-11-13 16:15:45.000000000 +0100
@@ -52,6 +52,11 @@
   #include <string.h>
   #include <float.h>
 
+#if !defined _MSC_VER && !defined __BORLANDC__
+  #define __STDC_CONSTANT_MACROS // avformat needs INT64_C from stdint which needs this, avformat is included in otherlibs/highgui/cvcap_ffmpeg.cpp
+  #include <stdint.h>
+#endif
+
   #if defined __ICL
     #define CV_ICC   __ICL
   #elif defined __ICC
@@ -141,8 +146,8 @@
 typedef __int64 int64;
 typedef unsigned __int64 uint64;
 #else
-typedef long long int64;
-typedef unsigned long long uint64;
+typedef int64_t int64;
+typedef uint64_t uint64;
 #endif
 
 #ifndef HAVE_IPL
