Index: opencv-1.0.0/cxcore/include/cxmisc.h
===================================================================
--- opencv-1.0.0.orig/cxcore/include/cxmisc.h	2009-11-14 17:03:05.000000000 +0100
+++ opencv-1.0.0/cxcore/include/cxmisc.h	2009-11-14 17:03:21.000000000 +0100
@@ -130,7 +130,7 @@
     #include <alloca.h>
 #elif defined HAVE_ALLOCA
     #include <stdlib.h>
-#else
+#elif
     #error
 #endif
 
