diff --git a/src/cv/CMakeLists.txt b/src/cv/CMakeLists.txt
index 5db832f..15b49d6 100644
--- a/src/cv/CMakeLists.txt
+++ b/src/cv/CMakeLists.txt
@@ -22,9 +22,13 @@ source_group("Include\\External" FILES ${lib_hdrs})
 file(GLOB lib_int_hdrs "*.h*")
 source_group("Include\\Internal" FILES ${lib_int_hdrs})
 
+set(need_files ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs})
+
 set(the_target "cv")
+set(the_target_s "cv_s")
 
-add_library(${the_target} ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs})
+add_library(${the_target} SHARED ${need_files})
+add_library(${the_target_s} STATIC ${need_files})
 
 if(PCHSupport_FOUND)
     set(pch_header ${CMAKE_CURRENT_SOURCE_DIR}/_cv.h)
@@ -45,6 +49,9 @@ set_target_properties(${the_target} PROPERTIES
     OUTPUT_NAME "${the_target}${OPENCV_DLLVERSION}"
     )
 
+# For static link library
+set_target_properties(${the_target_s} PROPERTIES OUTPUT_NAME ${the_target})
+
 # Additional target properties
 set_target_properties(${the_target} PROPERTIES
     DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
@@ -69,7 +76,7 @@ endif()
 # Dependencies of this target:
 add_dependencies(${the_target} cxcore)
 
-install(TARGETS ${the_target}
+install(TARGETS ${the_target} ${the_target_s}
     RUNTIME DESTINATION bin COMPONENT main
     LIBRARY DESTINATION lib COMPONENT main
     ARCHIVE DESTINATION lib COMPONENT main)
diff --git a/src/cvaux/CMakeLists.txt b/src/cvaux/CMakeLists.txt
index fc4edc4..2acb3db 100644
--- a/src/cvaux/CMakeLists.txt
+++ b/src/cvaux/CMakeLists.txt
@@ -32,9 +32,14 @@ include_directories("${CMAKE_CURRENT_SOURCE_DIR}/../../include/opencv"
 # ----------------------------------------------------------------------------------
 # 				Define the library target:
 # ----------------------------------------------------------------------------------
+
+set(need_files ${lib_srcs} ${lib_hdrs})
+
 set(the_target "cvaux")
+set(the_target_s "cvaux_s")
 
-add_library(${the_target} ${lib_srcs} ${lib_hdrs})
+add_library(${the_target} SHARED ${need_files})
+add_library(${the_target_s} STATIC ${need_files})
 
 if(PCHSupport_FOUND)
     set(pch_header ${CMAKE_CURRENT_SOURCE_DIR}/_cvaux.h)
@@ -55,6 +60,9 @@ set_target_properties(${the_target} PROPERTIES
 	OUTPUT_NAME "${the_target}${OPENCV_DLLVERSION}"
 	)
 
+# For static link library
+set_target_properties(${the_target_s} PROPERTIES OUTPUT_NAME ${the_target})
+
 # Additional target properties
 set_target_properties(${the_target} PROPERTIES
 	DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
@@ -74,7 +82,7 @@ endif()
 # Dependencies of this target:
 add_dependencies(${the_target} cxcore cv highgui ml)
 
-install(TARGETS ${the_target}
+install(TARGETS ${the_target} ${the_target_s}
 	RUNTIME DESTINATION bin COMPONENT main
 	LIBRARY DESTINATION lib COMPONENT main
 	ARCHIVE DESTINATION lib COMPONENT main)
diff --git a/src/cxcore/CMakeLists.txt b/src/cxcore/CMakeLists.txt
index e13bf32..6017d01 100644
--- a/src/cxcore/CMakeLists.txt
+++ b/src/cxcore/CMakeLists.txt
@@ -22,9 +22,13 @@ source_group("Include\\External" FILES ${lib_hdrs})
 file(GLOB lib_int_hdrs "*.h*")
 source_group("Include\\Internal" FILES ${lib_int_hdrs})
 
+set(need_files ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs})
+
 set(the_target "cxcore")
+set(the_target_s "cxcore_s")
 
-add_library(${the_target} ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs})
+add_library(${the_target} SHARED ${need_files})
+add_library(${the_target_s} STATIC ${need_files})
 
 if(PCHSupport_FOUND)
     set(pch_header ${CMAKE_CURRENT_SOURCE_DIR}/_cxcore.h)
@@ -45,6 +49,9 @@ set_target_properties(${the_target} PROPERTIES
     OUTPUT_NAME "${the_target}${OPENCV_DLLVERSION}"
     )
 
+# For static link library
+set_target_properties(${the_target_s} PROPERTIES OUTPUT_NAME ${the_target})
+
 # Additional target properties
 set_target_properties(${the_target} PROPERTIES
     DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
@@ -69,7 +76,7 @@ endif()
 
 add_dependencies(${the_target} opencv_lapack zlib)
 
-install(TARGETS ${the_target}
+install(TARGETS ${the_target} ${the_target_s}
     RUNTIME DESTINATION bin COMPONENT main
     LIBRARY DESTINATION lib COMPONENT main
     ARCHIVE DESTINATION lib COMPONENT main)
diff --git a/src/highgui/CMakeLists.txt b/src/highgui/CMakeLists.txt
index 19b93e4..e6946ff 100644
--- a/src/highgui/CMakeLists.txt
+++ b/src/highgui/CMakeLists.txt
@@ -183,7 +183,11 @@ set(lib_srcs ${highgui_srcs} ${grfmt_srcs})
 # ----------------------------------------------------------------------------------
 # 				Define the library target:
 # ----------------------------------------------------------------------------------
+
+set(need_files ${lib_srcs} ${highgui_hdrs} ${grfmt_hdrs} ${highgui_ext_hdrs})
+
 set(the_target "highgui")
+set(the_target_s "highgui_s")
 
 add_definitions(-DHIGHGUI_EXPORTS -DCVAPI_EXPORTS)
 
@@ -191,7 +195,8 @@ include_directories("${CMAKE_CURRENT_SOURCE_DIR}/../../include/opencv"
                     "${CMAKE_CURRENT_SOURCE_DIR}"
                     "${CMAKE_CURRENT_BINARY_DIR}")
 
-add_library(${the_target} ${lib_srcs} ${highgui_hdrs} ${grfmt_hdrs} ${highgui_ext_hdrs})
+add_library(${the_target} SHARED ${need_files})
+add_library(${the_target_s} STATIC ${need_files})
 
 if(PCHSupport_FOUND)
     set(pch_header ${CMAKE_CURRENT_SOURCE_DIR}/_highgui.h)
@@ -212,6 +217,9 @@ set_target_properties(${the_target} PROPERTIES
     OUTPUT_NAME "${the_target}${OPENCV_DLLVERSION}"
     )
 
+# For static link library
+set_target_properties(${the_target_s} PROPERTIES OUTPUT_NAME ${the_target})
+
 # Additional target properties
 set_target_properties(${the_target} PROPERTIES
 	DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
@@ -280,7 +288,7 @@ if(APPLE)
 	endif()
 endif()
 
-install(TARGETS ${the_target}
+install(TARGETS ${the_target} ${the_target_s}
 	RUNTIME DESTINATION bin COMPONENT main
 	LIBRARY DESTINATION lib COMPONENT main
 	ARCHIVE DESTINATION lib COMPONENT main)
diff --git a/src/ml/CMakeLists.txt b/src/ml/CMakeLists.txt
index ef6279c..31be9f7 100644
--- a/src/ml/CMakeLists.txt
+++ b/src/ml/CMakeLists.txt
@@ -16,9 +16,13 @@ include_directories("${CMAKE_CURRENT_SOURCE_DIR}/../../include/opencv"
                     "${CMAKE_CURRENT_SOURCE_DIR}"
                     "${CMAKE_CURRENT_BINARY_DIR}")
 
+set(need_files ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs})
+
 set(the_target "ml")
+set(the_target_s "ml_s")
 
-add_library(${the_target} ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs})
+add_library(${the_target} SHARED ${need_files})
+add_library(${the_target_s} STATIC ${need_files})
 
 if(PCHSupport_FOUND)
     set(pch_header ${CMAKE_CURRENT_SOURCE_DIR}/_ml.h)
@@ -39,6 +43,9 @@ set_target_properties(${the_target} PROPERTIES
     OUTPUT_NAME "${the_target}${OPENCV_DLLVERSION}"
     )
 
+# For static link library
+set_target_properties(${the_target_s} PROPERTIES OUTPUT_NAME ${the_target})
+
 # Additional target properties
 set_target_properties(${the_target} PROPERTIES
 	DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
@@ -59,7 +66,7 @@ target_link_libraries(${the_target} ${OPENCV_LINKER_LIBS} cxcore)
 # Dependencies of this target:
 add_dependencies(${the_target} cxcore)
 
-install(TARGETS ${the_target}
+install(TARGETS ${the_target} ${the_target_s}
 	RUNTIME DESTINATION bin COMPONENT main
 	LIBRARY DESTINATION lib COMPONENT main
 	ARCHIVE DESTINATION lib COMPONENT main)
